sympy/stats/tests/test_compound_rv.py

```diff
@@ -63,20 +63,33 @@ def test_unevaluated_CompoundDist():
     (_k, 0, oo))/(32*sqrt(pi)), True))
     assert (density(X)(x).simplify()).dummy_eq(exprd.simplify())
 
-    expre = Integral(Piecewise((_k*exp(S(3)/4 - _k/4)/8, 2*Abs(arg(_k - 3)) <= pi/2),
-    (sqrt(2)*_k*Integral(exp(-(_k**4 + 16*(_k - 3)**2)/(32*_k**2)),
-    (_k, 0, oo))/(32*sqrt(pi)), True)), (_k, -oo, oo))
+    expre = Integral(_k*Integral(sqrt(2)*exp(-_k**2/32)*exp(-(_k - 3)**2/(2*_k**2)
+    )/(32*sqrt(pi)), (_k, 0, oo)), (_k, -oo, oo))
     with ignore_warnings(UserWarning): ### TODO: Restore tests once warnings are removed
-        assert (E(X, evaluate=False).rewrite(Integral).simplify()).dummy_eq(expre.simplify())
-
+        assert E(X, evaluate=False).rewrite(Integral).dummy_eq(expre)
+
+    X = Poisson('X', 1)
+    Y = Poisson('Y', X)
+    Z = Poisson('Z', Y)
+    exprd = exp(-1)*Sum(exp(-Y)*Y**x*Sum(exp(-X)*X**Y/(factorial(X)*factorial(Y)
+                ), (X, 0, oo)), (Y, 0, oo))/factorial(x)
+    assert density(Z)(x).simplify() == exprd
+
+    N = Normal('N', 1, 2)
+    M = Normal('M', 3, 4)
+    D = Normal('D', M, N)
+    exprd = Integral(sqrt(2)*exp(-(_k - 1)**2/8)*Integral(exp(-(-_k + x
+    )**2/(2*_k**2))*exp(-(_k - 3)**2/32)/(8*pi*_k)
+    , (_k, -oo, oo))/(4*sqrt(pi)), (_k, -oo, oo))
+    assert density(D, evaluate=False)(x).dummy_eq(exprd)
 
 def test_Compound_Distribution():
     X = Normal('X', 2, 4)
     N = NormalDistribution(X, 4)
     C = CompoundDistribution(N)
     assert C.is_Continuous
     assert C.set == Interval(-oo, oo)
-    assert C.pdf(x).simplify() == exp(-x**2/64 + x/16 - S(1)/16)/(8*sqrt(pi))
+    assert C.pdf(x, True).simplify() == exp(-x**2/64 + x/16 - S(1)/16)/(8*sqrt(pi))
```

Comment:
This way of passing `evaluate=True` is a bit unusual... most often it's passed to `kwargs` so that it's mandatory to specify the parameter, like in `C.pdf(x, evaluate=True)`.

But that's just a curiosity... this looks good to me anyway.

Reply:
Thanks, I will change that.

Reply:
Looks good to merge now?